discovery_engine_core/core/src/storage.rs

```diff
@@ -65,7 +70,11 @@ pub(crate) trait FeedScope {
     async fn fetch(&self) -> Result<Vec<ApiDocumentView>, Error>;
 
     // helper function. will be replaced later by move_from_stacks_to_feed
-    async fn store_documents(&self, documents: &[NewDocument]) -> Result<(), Error>;
+    async fn store_documents(
+        &self,
+        documents: &[NewDocument],
+        stack_ids: &HashMap<document::Id, stack::Id>,
```

Comment:
In future when the stacks use the db directly this will be a bit nicer as we will always insert only a batch of documents for the same stack.